content/dev_stories/playground_to_production_fruit_falling.md

```diff
@@ -0,0 +1,677 @@
+    robust code structure (multiple files, types, etc.). With the code 
+    already in a working state, I moved functionality out into separate 
+    files incrementally (camera creation into a new camera type, falling
+    objects into a new class, etc.), proving at every step that I hadn't
+    broken my code. Then, once all the functionality had been moved out
+    of `playground.ts`, I simply deleted that file and the folder 
+    containing it from my repo.
+1.  I cleaned out the unused dependencies from the NPM Package Template
+    repository.
+
+    This is another step I recommend for moving your new repo away from 
+    Playground practices and into production. In order to be able to 
+    support code copy-pasted out of the Playground, the NPM Package
+    Template's `app_package` by default includes a *lot* of dependencies,
+    only a subset of which are likely to be used by any particular 
+    codebase. Fortunately, it's easy to get rid of the unnecessary imports
+    here; all I had to do was open `app_package/package.json` and delete
+    the references to any NPM packages I wasn't using--which, in this case,
+    was most of them.
+
+    ![Excess NPM package deletion](/img/devVignettes/fruitFalling/17_npm_package_deletion.jpg!500)
+1.  To make sure I was rendering under the intended circumstances, I 
+    customized `index.js` in the `test_package`.
+
+    The test app included with the NPM Package Template has an extremely
+    simplisitic and generic appearance: it's basically just a small 
+    `HTMLCanvasElement` in a blank page. Not all Web apps are designed for
+    generic canvases, however, so in many cases it may be helpful to add
+    features to the test app to better reflect the real environment where
+    the app will be run (full-screen capability, website forms to enter 
+    data, etc.). In my case, *Fruit Fallin'* was designed primarily with 
+    mobile touchscreens in mind, so I wanted my canvas to have an aspect 
+    ratio that more closely resembled what the game was ultimately supposed
+    to run in.
+
+    ![index.js in test_package](/img/devVignettes/fruitFalling/18_index_js_in_test_package.jpg!600)
+1.  I used `Scene`s as a unit of organization for my game logic. 
+
+    ![Scenes used in runtime.ts](/img/devVignettes/fruitFalling/19_scenes_in_runtime_ts.jpg!400)
+
+    Most interactive apps have multiple "modes" of operation, and *Fruit 
+    Fallin'* in particular has two: the game and the 
+    title/menu/options/credits. While there's no real rule for how best
+    to separate the concerns of the different modes, I found it useful to 
+    have them as separate Babylon `Scene` objects. Specifically, I made new
+    `Scene` subclasses with the behavior I wanted for the different "modes";
+    that made it easy for me to transition between modes by simply 
+    disposing one kind of scene and creating a new instance of the other, 
+    thus keeping the logic and resources clean and separated.

```

Comment:
The Scene subclassing approach is a great idea! I think showing parts of the code of one of these subclasses to exemplify this approach could be nice, as I expect less people are going to actually look at the codebase by itself, so we'd want to have as much upfront information as possible here. Another option would be linking the classes here like the fixed framerate observable example below.

Reply:
Linking the files is a great idea, I'll definitely do that. Ultimately, I want to revisit practices like this more thoroughly in later Stories when there's more of a coding focus, as going too deep on practices like this in this Story would make it even longer and distract from the "Playground to Production" focus. The linking to the source strikes a really good balance.